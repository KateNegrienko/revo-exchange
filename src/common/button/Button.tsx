import { FC, useCallback } from "react";
import cn from "classnames";
import { IButtonProps } from "./Button.interface";
import theme from "./Button.module.scss";

const Button: FC<IButtonProps> = ({
  onClick,
  className,
  children,
  disabled = false,
}) => {
  const handleClick = useCallback(() => {
    if (!disabled) {
      onClick();
    }
  }, [disabled, onClick]);
  return (
    <button
      className={cn(theme.button, disabled && theme.disabled, className)}
      onClick={handleClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

export default Button;
