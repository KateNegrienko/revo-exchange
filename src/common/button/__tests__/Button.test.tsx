import React from "react";
import { render, shallow } from "enzyme";
import Button from "../Button";

describe("[Button snapshots]", () => {
  it("Should render properly with only required props", () => {
    const children = <div>New Button</div>;
    const onClick = () => {};
    expect(
      render(<Button onClick={onClick}>{children}</Button>)
    ).toMatchSnapshot();
  });

  it("Should call appropriate handler when onClick event emits", () => {
    const mockedOnClick = jest.fn();

    const wrap = shallow(<Button onClick={mockedOnClick}>Ok</Button>);

    wrap.simulate("click");

    expect(mockedOnClick).toHaveBeenCalledTimes(1);
  });

  it("Should not call appropriate handler when onClick event emits on disabled button", () => {
    const mockedOnClick = jest.fn();
    const wrap = shallow(
      <Button disabled onClick={mockedOnClick}>
        Ok
      </Button>
    );

    wrap.simulate("click");

    expect(mockedOnClick).toHaveBeenCalledTimes(0);
  });
});
