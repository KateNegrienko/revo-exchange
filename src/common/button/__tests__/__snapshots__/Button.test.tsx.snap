// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button snapshots] Should render properly with only required props 1`] = `
<button
  class="button"
>
  <div>
    New Button
  </div>
</button>
`;
